src/os/bluestore/BlueStore.cc

```diff
@@ -3642,11 +3643,12 @@ void BlueStore::Onode::put() {
     // should be the last action since Onode can be released
     // at any point after this decrement
     if (need_unpin) {
```

Comment:
Lets update comments.
Onode can no longer be released at any point.

Reply:
And also comments here.
https://github.com/ceph/ceph/pull/43770/files#diff-6f3b1a5dcf5313dc06f71ba4997ee28ec219ecb5633a728b916465c684f1d87dR3638

Reply:
@aclamk - could you please elaborate on the second comment - I can find anything related under the link....

Reply:
@aclamk - and it looks like we can now remove that trick with an additional increment for pinned entries. This patch makes that safe. I'm inclined to submit that in a different patch though.
 Just asking for your thought for now

Reply:
@ifed01 The comment I meant was "// remove will also decrement nref and delete Onode". And It seems I posted broken link, or rebase moved it or something...
In this new situation, there is no way Onode could be deleted there.
"// remove will decrement nref, but Onode will not be deleted yet (put_nref > 1)"

Reply:
@ifed01
Regarding removal of artificial reference counter for pinning.
I do not believe we can to that.
BUT:
I look forward to be wrong in this case!

Reply:
> @ifed01 The comment I meant was "// remove will also decrement nref and delete Onode". And It seems I posted broken link, or rebase moved it or something... In this new situation, there is no way Onode could be deleted there. "// remove will decrement nref, but Onode will not be deleted yet (put_nref > 1)"

OK, thanks. Updated.

Reply:
> @ifed01 Regarding removal of artificial reference counter for pinning. I do not believe we can to that. BUT: I look forward to be wrong in this case!

See the last commit of https://github.com/ceph/ceph/pull/44311
Looks good so far for my overnight QA run.